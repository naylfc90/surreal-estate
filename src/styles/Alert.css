.alert-success {
  padding: 5px;
  background-color: rgba(0, 255, 0, 0.2);
  padding-left: 10px;
  width: 30%;
  border: 1px solid green;
  color: green;
}

.alert-error {
  padding: 5px;
  background-color: rgba(255, 0, 0, 0.2);
  padding-left: 10px;
  width: 30%;
  border: 1px solid red;
  color: red;
}
