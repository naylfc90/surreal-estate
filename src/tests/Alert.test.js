import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

test("error message is displayed", () => {
  const { getByText } = render(<Alert message="Error" />);

  expect(getByText(/Error/).textContent).toBe("Error");
});

test("success message is displayed", () => {
  const { getByText } = render(<Alert message="Success" />);

  expect(getByText(/Success/).textContent).toBe("Success");
});

test("does not render success/error message if props is empty", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});

test("error message is rendered if props is set to 'error'", () => {
  const { asFragment } = render(<Alert message="Error" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});

test("success message is rendered if props is set to 'success'", () => {
  const { asFragment } = render(<Alert message="Success" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});
