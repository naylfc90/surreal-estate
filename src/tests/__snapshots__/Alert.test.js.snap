// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render success/error message if props is empty 1`] = `<DocumentFragment />`;

exports[`error message is rendered if props is set to 'error' 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error
  </div>
</DocumentFragment>
`;

exports[`success message is rendered if props is set to 'success' 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Success
  </div>
</DocumentFragment>
`;
